variable/type_conversion.cpp

```diff
@@ -39,7 +39,9 @@ struct MakeVariableWithType {
   }
 };
 
-Variable astype(const Variable &var, DType type) {
-  return type == var.dtype() ? var : MakeVariableWithType::make(var, type);
+Variable astype(const Variable &var, DType type, const CopyPolicy copy) {
+  return type == var.dtype() && copy == CopyPolicy::TryAvoid
+             ? var
+             : MakeVariableWithType::make(var, type);
```

Comment:
It has always bugged me that we are creating `N**2` instead of `N**2-N` conversion paths here. If we used `copy(var)` here and somehow change line 22 to exclude `T`...?

Reply:
Done